### ISSUE TITLE
Update the chisel-lang documentation for this project

### ISSUE THREAD

**Issue by chick** [ucb-bar/chisel-testers2#352]
An update on basic stuff
- backends
  - what they are
  - how to change them and why
- caching
- decoupled

**Comment by ekiwi** [comment]
I agree with the list, besides the _caching_ aspect, which for now is for experts only.

**Comment by chick** [comment]
Add to this filling out the ScalaDoc for the decoupled interface stuff. Currently the object are documented but not the methods that they contain.

**Comment by carlosedp** [comment]
Maybe start a "Cookbook" like doc aggregating the most used scenarios using chiseltest is a good idea.
